pkg/metrics/collector/sockstat_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestSockstats(t *testing.T) {
+	input := "sockets: used 270\n" +
+		"TCP: inuse 23 orphan 0 tw 10 alloc 52 mem 6\n" +
+		"UDP: inuse 3 mem 3\n" +
+		"UDPLITE: inuse 0\n" +
+		"RAW: inuse 0\n" +
+		"FRAG: inuse 0 memory 0"
+	input6 := "TCP6: inuse 14\n" +
+		"UDP6: inuse 1\n" +
+		"UDPLITE6: inuse 0\n" +
+		"RAW6: inuse 0\n" +
+		"FRAG6: inuse 0 memory 0"
+
+	ss, err := parseSockStats(strings.NewReader(input))
+	if err != nil {
+		t.Errorf("parseSockStats returns err %v, want nil", err)
+	}
+
+	ss6, _ := parseSockStats(strings.NewReader(input6))
+	if err != nil {
+		t.Errorf("parseSockStats6 returns err %v, want nil", err)
+	}
+
+	if ss.tcpInuse+ss6.tcpInuse != 37 {
+		t.Errorf("parseSockStats returns tcpInuse %v, want 37",
+			ss.tcpInuse+ss6.tcpInuse)
+	}
+	if ss.udpInuse+ss6.udpInuse != 4 {
+		t.Errorf("parseSockStats returns udpInuse %v, want 4",
+			ss.udpInuse+ss6.udpInuse)
+	}
+	if ss.tcpTw != 10 {
+		t.Errorf("parseSockStats returns tcpTw %v, want 10", ss.tcpTw)
+	}
+	if ss.mem != 9 {
+		t.Errorf("parseSockStats returns mem %v, want 9", ss.mem)
+	}
+}
+
+func TestSockstatsInvalidInput(t *testing.T) {
+	for _, tc := range []struct {
+		desc  string
+		input string
+	}{
+		{
+			desc:  "empty file",
+			input: "",
+		},
+		{
+			desc:  "empty line",
+			input: " ",
+		},
+		{
+			desc:  "missing value",
+			input: "TCP:",
+		},
+		{
+			desc:  "wrong format",
+			input: "TCP inuse 23",
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			t.Parallel()
+			_, err := parseSockStats(strings.NewReader(tc.input))
+			if err == nil {
+				t.Errorf("parseSockStats(%q) = _, nil, want error", tc.input)
+			}
+		})
+	}
+
+	if _, err := getSockStats("./fake"); err == nil {
```

Comment:
uses of the filesystem should be hermetic (create a temporary subdir for this test)

also, probably don't need to check this in the unit test

Reply:
Removed the test case.